README.md

```diff
@@ -13,128 +13,145 @@ Simple cluster and environment specific aware templating for kubernetes manifest
 
-## installation
+## Installation
 
 run this
 
-```
+```bash
 $ pip install --user --upgrade k8t
 ```
 
 **note**: k8t is not Python 2 compatible
 
-## usage
+### Completion
 
-### scaffolding
+Run the following and store the file in your distribution/OS specific spot
 
-Create a new project folder with a cluster directory and an empty defaults file
+bash:
 
-```
-$ k8t new project foobar
+```bash
+$ _K8T_COMPLETE=source k8t > k8t-completion.sh
 ```
 
-Create a new cluster
+zsh:
 
-```
-$ k8t new cluster A
+```zsh
+$ _K8T_COMPLETE=source_zsh k8t > k8t-completion.sh
 ```
 
-Create a new environment
+## Concepts
 
-```
-$ k8t new environment staging -c A
-$ k8t new environment production
+By combining those concepts you can quickly add completely new environments to your deployment pipeline just by
+modifying specializing values and sharing the rest.
 
-```
-Generate a new deployment template for cluster A (currently only `deployment` and `service` are supported)
+Check out our examples [here](examples/).
 
-```
-$ k8t new template deployment -c A
-```
+### *Clusters* and *Environments*
+
+k8t comes with a builtin framework for *clusters* and *environments* (e.g. production, staging). This came from the need to be able to deploy
+the same application over multiple clusters and in different environments with completely different setups and values.
+This idea is helped by the fact that k8t deep-merges values and configs, allowing easy variation through different
+stages of your application deployment.
 
-Specify prefixes for secrets
+Both *clusters* and *environments* are intentionally working the same way and can be used to add another degree of freedom when
+combined. *Environments* however are also available globally, meaning clusters can share environment specific
+configuration while specifying differences in those environments.
 
+### Templating
+
+Templating is supported via [Jinja](https://jinja.palletsprojects.com). k8t also comes with some additional
+[helper functions](#template-helper-functions) and a [validation function](#validate-templates) with verbose output to
+quickly verify the written templates.
+
+#### Template helper functions
+
+* `random_password(N: int)` - generate a random string of length N
+* `envvar(key: str, [default])` - get a value from any environment variable with optional default
+* `b64encode(value: str)` - encodes a value in base64 (usually required for secrets)
+* `b64decode(value: str)` - decodes a value from base64
+* `hashf(value: str, [method: str])` - hashes a given value (default using `sha256`)
+* `get_secret(key: str)` - provides a secret value from a given provider (see [here](#managing-secrets))
+
+## Usage
+
+### Scaffolding
+
+Create a new project folder with a cluster directory and an empty defaults file
+
+```bash
+$ k8t new project .
 ```
-$ k8t edit config --cluster A --environment staging
-secrets:
-  prefix: "staging/application"```
+
+Create a new cluster
+
+```bash
+$ k8t new cluster A
```

Comment:
I think you should use a more verbose name for the example. Instead of "A" something like "MyCluster" would be more readable, and for the next commands it makes the understandinf easier